### ISSUE TITLE
Github protocols config ignored

### ISSUE THREAD

**Issue by FlorinPopaCodes** [composer/composer#1152]
My problem: I'm behind an enterprise firewall and I can use the git protocol, but as the project gets larger the download time get bigger. The https or http could easily fix this problem.

In the documentation there's a reference at for http://getcomposer.org/doc/04-schema.md#config for a configurable option named _github-protocols_ which should allow me to configure my project to download over a https or http protocol with a fallback to git.

After I've changed the config.json I've noticed there's no change in the download speed and after some research I've found that this option is ignored and the option is hard set in GitDownloader.php:154.

After I've manually changed the array at line 154 to suit my config, the download times have improved as intended.

tl;dr The option _github-protocols_ in the config array is ignored and hard set in GitDownloader.php:154.
